### ISSUE TITLE
(Hapi) Replace Hapi with something else

### ISSUE THREAD

**Issue by RickWong** [RickWong/react-isomorphic-starterkit#69]
Hapi's API is evolving at a crazy pace and its documentation is lacking. I don't think it's the simplest web server anymore.

**Comment by paOol** [comment]
disagree, the only "breaking" change in the past year or so was v9 which excluded 3 plugins.

you just npm install them again, or replace those plugins with whatever you choose.

But I got my hapi upgraded by just adding

`
server.register([require('vision'), require('inert'), require('h2o2')], function (err) {`

**Comment by RickWong** [comment]
Honestly it's not much work, I know, and it's less "crazy" as I described above, but it's Hapi's project vision. I don't think their way of modularity is healthy in the long run.

Serving static files with Express:
http://expressjs.com/starter/static-files.html

Proxying API requests with Express:
https://gist.github.com/davemo/4027855

Popularity of Express vs Hapi:
https://www.npmjs.com/package/express
https://www.npmjs.com/package/hapi

Express is backed by StrongLoop which is owned by IBM.

**Comment by Dindaleon** [comment]
I disagree as well, hapi is very good for developing large applications. I am actually moving away from express and going into hapi.

**Comment by johnbrett** [comment]
Hi @RickWong, thanks for providing this start kit! I'd love to hear where hapi's documentation is lacking so it can be improved, and also out of curiosity, your thoughts on how hapi's modularity is unhealthy? 

I agree for this starter project it wouldn't be much effort to switch to express, but it's when the size of your application grows past the starter stage and you start to add authentication, validation, more routes etc, the structure hapi provides becomes more beneficial, so I'm much more in favour of sticking with hapi. Just my two cents!

**Comment by RickWong** [comment]
I have investigated Express and found that it too isn't ideal. What I mean by unhealthy modularity is that the dependencies bring in a lot of dependencies with each their own readme's and (differing) API's to learn, and so on... It's the kind of learning curve that grows steeper with every step. 

After lots of investigation, finally I've arrived at Koa. They have a great API and more importantly that API is consistent in the extra modules they provide. The next major update of react-isomorphic-starterkit will very probably be Koa-based.
